src/Database/Database.php

```diff
@@ -1461,8 +1461,12 @@ public function count($table, array $condition = [], array $params = [])
 
 		$row = $this->fetchFirst($sql, $condition);
 
-		// Ensure to always return either a "null" or a numeric value
-		return is_numeric($row['count']) ? (int)$row['count'] : $row['count'];
+		if (!isset($row['count'])) {
+			$this->logger->notice('Invalid count.', ['table' => $table, 'expression' => $expression, 'condition' => $condition_string, 'callstack' => System::callstack()]);
```

Comment:
I just realized that the `$row` value is not part of the logging. Could you add it to the log? it could be helpful for debugging.